linalg/sparsemat.cpp

```diff
@@ -174,9 +175,21 @@ SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph)
       }
 
       // We probably do not need to set the ownership flags here.
-      I.Reset(); I.SetHostPtrOwner(true);
-      J.Reset(); J.SetHostPtrOwner(true);
-      A.Reset(); A.SetHostPtrOwner(true);
+      if (mt != MemoryType::SIZE)
+      {
+         I.Reset(mt);
+         J.Reset(mt);
+         A.Reset(mt);
+      }
+      else
+      {
+         I.Reset();
+         J.Reset();
+         A.Reset();
+      }
```

Comment:
I don't understand why you need to call the `Reset` method, this seems wrong to me, you just called `New` to get some memory, and you `Reset` the memory right after?

Reply:
It looks like in this `else` block (for `if (mat.Finalized())`) `I`, `J`, and `A` are not `New`'d in any ways, it's the `Rows` member that is allocated and used, does look true to you?

Reply:
Thank you, all the `#ifdef` made me loose sight of what was happening.